### DISCUSSION TITLE
How to integrate Maxdos into IPxe menu based on UEFI pattern?

### DISCUSSION THREAD

**Discussion by jonelinux** [ipxe/ipxe#1109] [General]
Maxdos is a tool that runs in DOS mode. I need to integrate it into the IPxe menu to guide. 
IPxe runs in UEFI mode, and the IPxe menu loads IPxe.html through HTTP:

:maxdos
  initrd ${xieyi}://${next-server}/ipxe/maxdos.img
  kernel ${xieyi}://${next-server}/ipxe/memdisk initrd=maxdos.img
  boot

but it's not working，Can't DOS run in UEFI mode? Is there any other way to run it?
thanks!

![75ed5e5bc47422176f205aa2a64d722](https://github.com/ipxe/ipxe/assets/31331818/88e5f5fd-6e6d-424a-a6a9-fa1de092494e)

**Comment by NiKiZe** [comment]
Duplicate of #1108
Is this intentional or what is going on?

**Reply by jonelinux** [reply]
I didn't describe the https://github.com/ipxe/ipxe/discussions/1108 post very clearly, so I closed it and posted a new one

**Reply by NiKiZe** [reply]
You didn't even read or respond to the comments on #1108, update your post there delete this one!

But running DOS under EFI is not really a easy task since it relies on many INT calls only available in pcbios mode.
